ethereum_history_api/circuits/lib/src/log_test.nr

```diff
@@ -0,0 +1,89 @@
+mod extract_log {
+    use crate::fixtures::mainnet::{
+        cancun::{
+        small_block::{receipt::receipt_rlp as small_block_receipt_rlp, log::log as small_block_log},
+        approve::{receipt::receipt_rlp as approve_receipt_rlp, log::log as approve_log}
+    }
+    };
+    use crate::log::{extract_log, PhantomMaxLogDataLen, PhantomMaxLogsCount};
+    use crate::misc::arrays::sub_array_equals_up_to_length;
+
+    global phantom_max_log_data_len: PhantomMaxLogDataLen<32> = PhantomMaxLogDataLen {};
+    global phantom_max_logs_count: PhantomMaxLogsCount<8> = PhantomMaxLogsCount {};
+
+    #[test]
+    fn success_small_block() {
+        let log = extract_log(
+            small_block_receipt_rlp,
+            0,
+            phantom_max_log_data_len,
+            phantom_max_logs_count
+        );
+        assert_eq(log, small_block_log);
+    }
+
+    #[test]
+    fn success_approve() {
+        let log = extract_log(
+            approve_receipt_rlp,
+            0,
+            phantom_max_log_data_len,
+            phantom_max_logs_count
+        );
+        assert_eq(log, approve_log);
+    }
+
+    #[test]
+    fn big_max_log_data_len() {
+        let big_phantom_max_log_data_len: PhantomMaxLogDataLen<100> = PhantomMaxLogDataLen {};
+        let log = extract_log(
+            small_block_receipt_rlp,
+            0,
+            big_phantom_max_log_data_len,
+            phantom_max_logs_count
+        );
+        assert_eq(log.address, small_block_log.address);
+        assert_eq(log.topics, small_block_log.topics);
+        assert_eq(log.data.len(), small_block_log.data.len());
+        assert(
+            sub_array_equals_up_to_length(
+                small_block_log.data.storage,
+                log.data.storage,
+                0,
+                log.data.len()
+            )
+        );
+    }
+
```

Comment:
It would be probably better if we had separate tests for all the functions. Then we would have this test for `extract_log_fragment` function

Reply:
I consider those functions as private implementation detail, so I don't want expose them outside and make tests depend on a specific implementation. Furthermore, we have a fixtures for a rlp encoded receipt and final logs, testing intermediate methods will require to prepare a lot of new test data that potentially will change after every refactor.